ci/functional/job_cleanup.sh

```diff
@@ -23,7 +23,7 @@ fi
 
 arts="$arts$(ls ./*daos{,_agent}.log* 2>/dev/null)" && arts="$arts"$'\n'
 arts="$arts$(ls -d \
-   install/lib/daos/TESTING/ftest/avocado/job-results/job-* 2>/dev/null)" && \
+   install/lib/daos/TESTING/ftest/avocado/job-results/* 2>/dev/null)" && \
```

Comment:
It's necessary because the folders are no longer named job-*.
I thought lines 18-22 in existing script was supposed to remove sym link latest as the comment indicated but looks like it is not doing what it's suppose to.
Let me see what I can do to exclude the sym link.

Reply:
> I thought lines 18-22 in existing script was supposed to remove sym link latest as the comment indicated but looks like it is not doing what it's suppose to.

We can see that, if we look in a [log](https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-4069/runs/37/nodes/744/log/?start=0):

```
[2021-01-11T16:54:27.700Z] + '[' -e install/lib/daos/TESTING/ftest/avocado/job-results/latest ']'
[2021-01-11T16:54:27.700Z] ++ ls './*daos.log*' './*daos_agent.log*'
```

We can see no `rm` was done, so the symlink was not found at that time.  That should be debugged and fixed there.  The reason is that the separation (nominally at least) between `ftest.sh` and `launch.py` is that `ftest.sh` does things that are applicable for CI but might not be applicable for running locally.  Preserving `latest` in local runs seems like the reasonable thing to do, but we do want to get rid of it in CI.

Reply:
I will remove lines 18-22 from this script. Will take care of the sym link in launch.py

Reply:
> I will remove lines 18-22 from this script. Will take care of the sym link in launch.py

That's not the correct solution.  As I explained above, removing that symlink is only necessary (probably desired even) for CI, not for local running.  That is why it's done in `ftest.sh` which is generally CI's wrapper around `launch.py` to do things that are specific to running `launch.py` in CI.  It quite possible that when running locally, using `launch.py` developers will want the `latest` symlink preserved.  I know I do when I'm using `launch.py` locally.

So it needs to be figured out why the `latest` symlink in `ftest.sh` is not being detected.  Clearly the suspenders (belt and suspenders) of only archiving `job-*` directories has masked this failure to find and remove the `latest` symlink.  It's like just a path move/change that missed being updated when removing that.

Reply:
Did a few experiments in the script here, which worked locally on detecting the sym link and removing it but didn't work in CI. In the latest commit, I went back to removing the latest sym link at the end on the run in launch.py but with condition where it will only do it in CI. You will retain the sym link when run locally.

Reply:
Lines 18-22 are restored. The sym link 'latest' still shows in artifacts but doesn't affect anything that I'm aware of including result reporting. We can address this is a separate PR.

Reply:
This needs to be resolved as it causes failures to be able to restart a previous test result from a stage:

```
[2021-02-06T15:02:08.989Z] java.nio.file.NoSuchFileException: /var/lib/jenkins/jobs/daos-stack/jobs/daos/branches/PR-4470/builds/14/archive/Functional on CentOS 7/latest
```

Reply:
The suggestion from Phil above (with slight syntax modification from daosbuild) did not do the trick.